lib/covid19_orientation/data/store_pg.ex

```diff
@@ -0,0 +1,57 @@
+defmodule Covid19Orientation.Data.PgStore do
+  @moduledoc """
+  Store data in PostgreSQL
+
+  Test table:
+  ```sql
+  DROP TABLE IF EXISTS journal;
+  CREATE TABLE journal (
+      id_internal SERIAL PRIMARY KEY,
+      id VARCHAR,
+      timestamp TIMESTAMP WITH TIME ZONE,
+      data JSONB,
+      date_created TIMESTAMP WITH TIME ZONE DEFAULT now()
+  );
+  ```
+
+  """
+
+  use Agent
+
+  def start_link(_initial) do
+    Agent.start_link(
+      fn ->
+        {:ok, pid} =
+          Postgrex.start_link(
+            Application.fetch_env!(:covid19_orientation, __MODULE__)[:conn_opts]
+          )
+
+        %{pg: pid}
+      end,
+      name: __MODULE__
+    )
+  end
+
+  def write({timestamp, id}, data) when is_map(data) do
+    Agent.update(__MODULE__, fn state ->
+      write_to_pg(state.pg, timestamp, id, data)
+      state
+    end)
```

Comment:
```suggestion
      write_to_pg(state.pg, timestamp, id, data)
```
Je ne vois pas pourquoi on a besoin de mettre l'état de l'agent à jour

Reply:
Update/3 attend comme retour un nouvel état, quoiqu'il arrive. Cette modif écraserait l'état existant, provoquant le crash de l'application.

D'où l'idée d'utiliser cast/3, qui à en plus l'avantage d'être plus rapide, mais à voir comment celui-ci réagit si la connexion PG est perdue.

Reply:
Ce que je propose là c'est de ne pas utiliser du tout l'agent et juste se servir de son état pour récupérer le pid et le passer à write_to_pg.

Mais oui faire le cast c'est très bien aussi